site/js/modules/webui/course.js

```diff
@@ -35,12 +39,140 @@ function handlerCourse(path, params, context, container) {
+                        </div>
+                        <div class="input-field">
+                            <label for="email-cc">CC</label>
+                            <input type="text" id="email-cc" name="cc" />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-bcc">BCC</label>
+                            <input type="text" id="email-bcc" name="bcc" />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-subject">Subject</label>
+                            <input type="text" id="email-subject" name="subject" required />
+                        </div>
+                        <div class="input-field">
+                            <label for="email-body">Content</label>
+                            <textarea id="email-body" name="body" rows="10"></textarea>
+                        </div>
+                        <div class="checkbox-field">
+                            <input type="checkbox" id="dry-run" name="dry-run" />
+                            <label for="dry-run">Send as Dry Run</label>
+                        </div>
+                        <div class="checkbox-field">
+                            <input type="checkbox" id="html-format" name="html-format" />
+                            <label for="html-format">Content is in HTML</label>
+                        </div>
+                    </fieldset>
+                </div>
+                <button class="send-email">Send Email</button>
+                <div class="results-area"></div>
+            </div>
+        </div>
+    `;
+
+    document.querySelector('button.send-email').addEventListener('click', function(event) {
+        Routing.loadingStart(container.querySelector(".results-area"), false);
+
+        let args = {
+            [Routing.PARAM_COURSE_ID]: course.id,
+            [Routing.PARAM_EMAIL_TO]: extractRecipients(container.querySelector("fieldset [name='to']").value),
+            [Routing.PARAM_EMAIL_CC]: extractRecipients(container.querySelector("fieldset [name='cc']").value),
+            [Routing.PARAM_EMAIL_BCC]: extractRecipients(container.querySelector("fieldset [name='bcc']").value),
+            [Routing.PARAM_EMAIL_SUBJECT]: container.querySelector("fieldset [name='subject']").value,
+            [Routing.PARAM_EMAIL_BODY]: container.querySelector("fieldset [name='body']").value,
+            [Routing.PARAM_DRY_RUN]: container.querySelector("fieldset [name='dry-run']").checked,
+            [Routing.PARAM_EMAIL_HTML]: container.querySelector("fieldset [name='html-format']").checked,
+        };
+
+        if (args[Routing.PARAM_EMAIL_TO].length === 0 &&
+                args[Routing.PARAM_EMAIL_CC].length === 0 &&
+                args[Routing.PARAM_EMAIL_BCC].length === 0) {
+            renderResult('<p>Please specify at least one recipient.</p>');
```

Comment:
You still have some extra "please"s.
Make sure to keep in mind that every comment I make can apply to the entire PR (especially style issues).